airflow/contrib/hooks/hortonworks_ambari_hook.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+import json
+import time
+
+from airflow.hooks.http_hook import HttpHook
+from airflow.exceptions import AirflowException
+
+
+class HdpAmbariHook(HttpHook):
+
+    def __init__(self, ambari_conn_id='hortonworks_ambari_default'):
+        """
+
+        :param azure_conn_id: connection id of Hortonworks Ambari cluster
+                    which will be used to Hortonworks rest service
+        :type ambari_conn_id: str
+        """
+        super(HdpAmbariHook, self).__init__(ambari_conn_id)
+
+        self.headers = {
+            'Content-Type': "application/x-www-form-urlencoded"
```

Comment:
It would be nice to support basic and token authentication with Extra section available information. 
In addition to this it would be nice to add a "X-Requested-By: username" in case CSRF is not disabled in Ambari.